oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/core/OAuth2TokenIntrospectionClaimAccessor.java

```diff
@@ -130,14 +130,25 @@ default String getSubject() {
 	}
 
 	/**
-	 * Returns the issuer {@code (iss)} of the token
-	 * @return the issuer of the token
+	 * Returns the issuer {@code (iss)} of the token as a URL.
+	 * @deprecated use {@link #getIssuerName} instead, as some issuers may not be valid
+	 * URLs
+	 * @return the issuer of the token as a URL
 	 */
 	@Nullable
+	@Deprecated
 	default URL getIssuer() {
 		return getClaimAsURL(OAuth2TokenIntrospectionClaimNames.ISS);
 	}
 
+	/**
+	 * Returns the issuer {@code (iss)} of the token as a string.
+	 * @return the issuer of the token as a String
+	 */
+	default String getIssuerName() {
```

Comment:
Let's please leave this out for the time being. Applications can do:

```java
String issuer = accessor.getClaim("iss");
```

Reply:
Given that `JwtClaimAccessor` does the same thing, there may be some more discussion around this, and I'd like to be consistent. Feel free to open a ticket to discuss further.

Reply:
@jzheaux sure thing, will revert.

Reply:
@jzheaux sure thing, will revert.

Reply:
Reverted.